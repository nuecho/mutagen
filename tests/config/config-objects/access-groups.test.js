const checkKey = (cfgObject) => {return cfgObject.group.name === "cfgObjectTest"};

cfgObjectTests("accessGroups", {checkKey: checkKey, checkMandatoryProperties: false, checkUnchangeableProperties: true});
