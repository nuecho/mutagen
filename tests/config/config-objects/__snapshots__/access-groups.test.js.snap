// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen accessGroups config import-export] should create the accessGroups: the accessGroups have been created 1`] = `
Object {
  "stderr": "",
  "stdout": "+ accessgroup [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen accessGroups config import-export] should skip the same accessGroups: the accessGroups have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "= accessgroup [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen accessGroups config import-export] should throw exception when updating accessGroups with different values for unchangeable properties: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Unchangeable properties:
  - accessgroup [Environment/cfgObjectTest]:
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen accessGroups config import-export] should update the modified accessGroups: the modified accessGroups have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ accessgroup [Environment/cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;
